clippy_lints/src/missing_inline.rs

```diff
@@ -173,3 +163,23 @@ impl<'tcx> LateLintPass<'tcx> for MissingInline {
         check_missing_inline_attrs(cx, attrs, impl_item.span, desc);
     }
 }
+
+fn span_contains_extern(cx: &LateContext<'_>, span: Span) -> bool {
+    let Some(snippet) = span.get_source_text(cx) else {
+        return false;
+    };
+
+    for (kind, ident, _) in tokenize_with_text(&snippet) {
+        if kind != TokenKind::Ident {
+            continue;
+        }
+
+        match ident {
+            "extern" => return true,
+            "fn" => break,
+            _ => {},
+        }
+    }
+
+    false
+}
```

Comment:
Rather than retokenizing spans, can we just copy what rustc is doing and invert the condition?
https://github.com/rust-lang/rust/blob/1079c5edb2bd837e5c4cf8c7db2892db359a3862/compiler/rustc_passes/src/check_attr.rs#L591-L600